src/peft/utils/other.py

```diff
@@ -202,7 +202,15 @@ def check_module(self):
         # ModuleList, even though their forward methods cannot be called
         forbidden_classes = (torch.nn.ModuleDict, torch.nn.ModuleList, torch.nn.ParameterDict, torch.nn.ParameterList)
         if isinstance(self.original_module, forbidden_classes):
-            cls_name = self.original_module.__class__.__name__
+            cls_name = self.original_module.__class__
```

Comment:
Why discard `__name__`?

Reply:
This is what I referred to above:

> the error message inside of
> `ModulesToSaveWrapper.check_module` has been slightly adjusted. Previously, the class name would be used, which can be confusing. E.g. for LoRA, the class name of the linear LoRA layer is just `"Linear"`, which looks the same as `nn.Linear`. Therefore, the full name is now shown.

So the error message used to be:

> 'modules_to_save cannot be applied to modules of type Linear'

which is very confusing, as the user has no idea that "Linear" refers to PEFT's LoRA Linear layer. Now the message would state:

> modules_to_save cannot be applied to modules of type <class 'peft.tuners.lora.layer.Linear'>"